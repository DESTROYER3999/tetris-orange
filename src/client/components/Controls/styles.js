import styled from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  width: 70%;
  div {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    kbd {
      font-size: 2vh;
      border-radius: 3px;
      padding: 1px 2px 0;
      border: 3px solid white;
    }
    span {
      font-size: 1.5vh;
    }
  }
`;
