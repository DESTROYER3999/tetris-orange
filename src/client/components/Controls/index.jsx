import React from "react";
import { StyledContainer } from "./styles";

const Controls = () => {
  return (
    <StyledContainer>
      <div id="key-up">
        <kbd className="key">up</kbd>
        <span>piece rotation</span>
      </div>
      <div id="key-down">
        <kbd className="key">down</kbd>
        <span>move piece down</span>
      </div>
      <div id="key-left">
        <kbd className="key">left</kbd>
        <span>move piece left</span>
      </div>
      <div id="key-right">
        <kbd className="key">right</kbd>
        <span>move piece right</span>
      </div>
      <div id="key-space">
        <kbd className="key">space</kbd>
        <span>hard drop</span>
      </div>
    </StyledContainer>
  );
};

export default Controls;
