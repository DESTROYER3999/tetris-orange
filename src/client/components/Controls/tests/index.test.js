import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Controls from "../index";

describe("<Controls />", () => {
  test("Controls rendering without crashing", () => {
    const wrapper = shallow(<Controls />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
