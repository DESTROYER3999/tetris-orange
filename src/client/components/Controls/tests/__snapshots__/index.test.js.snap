// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> Controls rendering without crashing 1`] = `
<styled.div>
  <div
    id="key-up"
  >
    <kbd
      className="key"
    >
      up
    </kbd>
    <span>
      piece rotation
    </span>
  </div>
  <div
    id="key-down"
  >
    <kbd
      className="key"
    >
      down
    </kbd>
    <span>
      move piece down
    </span>
  </div>
  <div
    id="key-left"
  >
    <kbd
      className="key"
    >
      left
    </kbd>
    <span>
      move piece left
    </span>
  </div>
  <div
    id="key-right"
  >
    <kbd
      className="key"
    >
      right
    </kbd>
    <span>
      move piece right
    </span>
  </div>
  <div
    id="key-space"
  >
    <kbd
      className="key"
    >
      space
    </kbd>
    <span>
      hard drop
    </span>
  </div>
</styled.div>
`;
